diff -x *.git -r -u ./src/systems/sdl/sdl_music.cc ../rlvm-mod/src/systems/sdl/sdl_music.cc
--- ./src/systems/sdl/sdl_music.cc	Wed Mar 25 06:08:15 2020
+++ ../rlvm-mod/src/systems/sdl/sdl_music.cc	Sat Mar 28 08:44:28 2020
@@ -148,6 +148,7 @@
 
   int count;
   if (!s_bgm_enabled || !music || music->music_paused_) {
+    stream = (Uint8*) malloc(len);
     memset(stream, 0, len);
     return;
   }
@@ -179,6 +180,7 @@
     if (music->fade_count_ > count_total) {
       music->loop_point_ = STOP_NOW;
       s_currently_playing.reset();
+      stream = (Uint8*) malloc(len);
       memset(stream, 0, len);
       return;
     }
diff -x *.git -r -u ./vendor/xclannad/nwatowav.cc ../rlvm-mod/vendor/xclannad/nwatowav.cc
--- ./vendor/xclannad/nwatowav.cc	Wed Mar 25 06:08:16 2020
+++ ../rlvm-mod/vendor/xclannad/nwatowav.cc	Fri Mar 27 21:39:17 2020
@@ -569,7 +569,7 @@
 	if (skip_count) {
 		int skip_c = skip_count * channels * (bps/8);
 		retsize -= skip_c;
-		memmove(data, data+skip_c, skip_c);
+		// memmove(data, data+skip_c, skip_c);
 		skip_count = 0;
 	}
 	curblock++;
